presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -110,6 +116,26 @@ public void testNonComparableWindowOrder()
         assertFails(TYPE_MISMATCH, "SELECT row_number() OVER (ORDER BY t.x) FROM (VALUES(color('red'))) AS t(x)");
     }
 
+    @Test
+    public void testORWarning()
```

Comment:
What about IN conditions? These two statements are equivalent, but we would throw a warning for one and not the other:
`SELECT * FROM t1 JOIN t2 ON t1.a = t2.a OR t1.a = t2.b`
`SELECT * FROM t1 JOIN t2 ON t1.a IN (t2.a, t2.b)`

Reply:
We can add this as a separate rule. So for now it's fine.